scipy/spatial/ckdtree/ckdtree.pyx

```diff
@@ -1,19 +1,75 @@
 # Copyright Anne M. Archibald 2008
-# Additional contributions by Patrick Varilly and Sturla Molden
+# Additional contributions by Patrick Varilly and Sturla Molden 2012
+# Revision by Sturla Molden 2015
+# Balanced kd-tree construction written by Jake Vanderplas for scikit-learn
 # Released under the scipy license
+
+# distutils: language = c++
+
 import numpy as np
 import scipy.sparse
 
 cimport numpy as np
-cimport libc.stdlib as stdlib
+    
+from cpython.mem cimport PyMem_Malloc, PyMem_Realloc, PyMem_Free
+
 cimport cython
 
+from multiprocessing import cpu_count
+import threading
+
 cdef extern from "limits.h":
     long LONG_MAX
-cdef np.float64_t infinity = np.inf
+    
+cdef extern from "ckdtree_cpp_methods.h":
+    int number_of_processors
+    np.float64_t infinity
+    np.float64_t dmax(np.float64_t x, np.float64_t y)
+    np.float64_t dabs(np.float64_t x)
+    np.float64_t _distance_p(np.float64_t *x, np.float64_t *y,
+                                     np.float64_t p, np.intp_t k,
+                                     np.float64_t upperbound)
+infinity = np.inf
+number_of_processors = cpu_count()
 
-__all__ = ['cKDTree']
+from libcpp.vector cimport vector
 
+__all__ = ['cKDTree', 'cKDTreeNode']
```

Comment:
This makes ``cKDTreeNode`` public. I think that's important enough to list in the summary of this PR.